cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -367,26 +367,45 @@ int VpiSignalObjHdl::set_signal_value_str(std::string &value)
     value_s.value.str = &writable[0];
     value_s.format = vpiStringVal;
 
-    return set_signal_value(value_s);
+    return set_signal_value(value_s, action);
 }
 
-int VpiSignalObjHdl::set_signal_value(s_vpi_value value_s)
+int VpiSignalObjHdl::set_signal_value(s_vpi_value value_s, gpi_set_action_t action)
 {
     FENTER
-
+    PLI_INT32 vpi_put_flag;
     s_vpi_time vpi_time_s;
 
     vpi_time_s.type = vpiSimTime;
     vpi_time_s.high = 0;
     vpi_time_s.low  = 0;
 
-    if (vpiStringVar == vpi_get(vpiType, GpiObjHdl::get_handle<vpiHandle>())) {
-        // assigning to a vpiStringVar only seems to work with vpiNoDelay
+    switch (action) {
+    case GPI_FORCE:
+      vpi_put_flag = vpiForceFlag;
+      break;
+    case GPI_RELEASE:
+      // Best to pass its current value to the sim when releasing
+      vpi_get_value(GpiObjHdl::get_handle<vpiHandle>(),&value_s);
+      vpi_put_flag = vpiReleaseFlag;
+      break;
+    default:
```

Comment:
Can you use the explicit GPI_DEPOSIT flag here, and put an `assert(0)` into the default block to catch us potential bugs when extending the enum?